### PR TITLE
Swap elasticsearch gem for searchkick

### LINKED ISSUE #1610: Search by alternative name doesn't find
Search for puha should find pūhā, because it is an alternative name. But it doesn't.

### PR DESCRIPTION
**Pull request by Br3nda**
the searchkick gem is simpler to use and has more features.
 It provides some out of the box support for features that the official client does not (autocomplete, spelling suggestions, stemming, etc.).

https://docs.bonsai.io/article/99-searchkick

this PR swaps out the elasticsearch gem for searchkick and deletes a lot of code as a result.

closes #1610

New features:
- [x] find by alternate names
- [x] scientific names
- [x] misspellings 
- [x] plurals 
- [x] biased to crops you've planted 
- [x] biased towards crops with higher planting counts

### CODE REVIEW

**Comment by Br3nda** [comment]
It's working. Getting way more matches now, and i think the indexing and searching code it clearer to understand.

**Comment by Br3nda** [comment]
That code climate fail is bogus. if you click through, there's no issues.

File: spec/services/crop_search_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        FactoryBot.create :alternate_name, name: 'fungus', crop: mushroom
+        # scientific name
+        FactoryBot.create :scientific_name, name: 'Agaricus bisporus', crop: mushroom
+
+        # Requested and rejected
+        FactoryBot.create(:rejected_crop, name: 'rejected mushroom')
+        FactoryBot.create(:crop_request, name: 'requested mushroom')
+
+        # Child record
+        FactoryBot.create(:crop, name: 'portabello', parent: mushroom)
```
Comment (pozorvlak) [inline spec/services/crop_search_service_spec.rb]:
```suggestion
        FactoryBot.create(:crop, name: 'portobello', parent: mushroom)
```

**Comment by pozorvlak** [comment]
The conflict in `Gemfile.lock` looks straightforward enough, but I'm reluctant to hand-edit a generated file: @Br3nda, can you run `bundle update` to fix the conflict and then merge the PR once it's green?

**Comment by Br3nda** [comment]
Added spec for biasing results to crops with higher planting counts, and crops the logged in member has planted

**Pull request merged**